torchreid/engine/builder.py

```diff
@@ -9,80 +10,135 @@ def build_engine(cfg, datamanager, model, optimizer, scheduler,
-            label_smooth=cfg.loss.softmax.label_smooth,
-            aug_type=cfg.loss.softmax.augmentations.aug_type,
-            aug_prob=cfg.loss.softmax.augmentations.aug_prob,
-            decay_power=cfg.loss.softmax.augmentations.fmix.decay_power,
-            alpha=cfg.loss.softmax.augmentations.alpha,
-            size=(cfg.data.height, cfg.data.width),
-            max_soft=cfg.loss.softmax.augmentations.fmix.max_soft,
-            reformulate=cfg.loss.softmax.augmentations.fmix.reformulate,
-            pr_product=cfg.loss.softmax.pr_product,
-            softmax_type=softmax_type,
-            m=cfg.loss.softmax.m,
-            s=cfg.loss.softmax.s,
-            compute_s=cfg.loss.softmax.compute_s,
-            margin_type=cfg.loss.softmax.margin_type,
-            end_s=cfg.loss.softmax.end_s,
-            duration_s=cfg.loss.softmax.duration_s,
-            skip_steps_s=cfg.loss.softmax.skip_steps_s,
-            enable_masks=cfg.data.enable_masks,
-            adaptive_margins=cfg.loss.softmax.adaptive_margins,
-            class_weighting=cfg.loss.softmax.class_weighting,
-            attr_cfg=cfg.attr_loss,
-            base_num_classes=cfg.loss.softmax.base_num_classes,
-            symmetric_ce=cfg.loss.softmax.symmetric_ce,
-            mix_weight=cfg.mixing_loss.enable * cfg.mixing_loss.weight,
-            enable_rsc=cfg.model.self_challenging_cfg.enable,
-            enable_sam=cfg.sam.enable,
-            should_freeze_aux_models=should_freeze_aux_models,
-            nncf_metainfo=nncf_metainfo,
-            initial_lr=initial_lr,
-            use_ema_decay=cfg.train.ema.enable,
-            ema_decay=cfg.train.ema.ema_decay
-        )
-    elif cfg.loss.name == 'contrastive':
-        engine = ImageContrastiveEngine(
-            datamanager,
-            model,
-            optimizer=optimizer,
-            reg_cfg=cfg.reg,
-            scheduler=scheduler,
-            use_gpu=cfg.use_gpu,
-            s=cfg.loss.softmax.s,
-            end_s=cfg.loss.softmax.end_s,
-            duration_s=cfg.loss.softmax.duration_s,
-            skip_steps_s=cfg.loss.softmax.skip_steps_s,
-        )
-    else:
-        engine = ImageTripletEngine(
-            datamanager,
-            model,
-            optimizer=optimizer,
-            margin=cfg.loss.triplet.margin,
-            weight_t=cfg.loss.triplet.weight_t,
-            weight_x=cfg.loss.triplet.weight_x,
-            scheduler=scheduler,
-            use_gpu=cfg.use_gpu,
-            label_smooth=cfg.loss.softmax.label_smooth,
-            conf_penalty=cfg.loss.softmax.conf_penalty
-        )
+    is_accuracy_aware_training = False
+        if nncf_metainfo is not None:
+            if nncf_metainfo.nncf_config.get('compression').get('accuracy_aware_training', None) is not None:
+                is_accuracy_aware_training = True
+
+                                      'softmax and am_softmax losses for data.type = image')
+    if cfg.data.type == 'image':
```

Comment:
we don't support other data types now and remove it from config

Reply:
Done